core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -449,81 +452,81 @@ func (o *evmTxStore) Transactions(offset, limit int) (txs []Tx, count int, err e
+	if err = o.q.SelectContext(ctx, &dbTxs, sql, limit, offset); err != nil {
 		return
 	}
 	txs = dbEthTxsToEvmEthTxs(dbTxs)
-	err = o.preloadTxAttempts(txs)
+	err = o.preloadTxAttempts(ctx, txs)
 	return
 }
 
 // TxAttempts returns the last tx attempts sorted by created_at descending.
-func (o *evmTxStore) TxAttempts(offset, limit int) (txs []TxAttempt, count int, err error) {
+func (o *evmTxStore) TxAttempts(ctx context.Context, offset, limit int) (txs []TxAttempt, count int, err error) {
 	sql := `SELECT count(*) FROM evm.tx_attempts`
-	if err = o.q.Get(&count, sql); err != nil {
+	if err = o.q.GetContext(ctx, &count, sql); err != nil {
 		return
 	}
 
 	sql = `SELECT * FROM evm.tx_attempts ORDER BY created_at DESC, id DESC LIMIT $1 OFFSET $2`
 	var dbTxs []DbEthTxAttempt
-	if err = o.q.Select(&dbTxs, sql, limit, offset); err != nil {
+	if err = o.q.SelectContext(ctx, &dbTxs, sql, limit, offset); err != nil {
 		return
 	}
 	txs = dbEthTxAttemptsToEthTxAttempts(dbTxs)
-	err = o.preloadTxesAtomic(txs)
+	err = o.preloadTxesAtomic(ctx, txs)
 	return
 }
 
 // FindTxAttempt returns an individual TxAttempt
-func (o *evmTxStore) FindTxAttempt(hash common.Hash) (*TxAttempt, error) {
+func (o *evmTxStore) FindTxAttempt(ctx context.Context, hash common.Hash) (*TxAttempt, error) {
 	dbTxAttempt := DbEthTxAttempt{}
 	sql := `SELECT * FROM evm.tx_attempts WHERE hash = $1`
-	if err := o.q.Get(&dbTxAttempt, sql, hash); err != nil {
+	if err := o.q.GetContext(ctx, &dbTxAttempt, sql, hash); err != nil {
 		return nil, err
 	}
 	// reuse the preload
 	var attempt TxAttempt
 	dbTxAttempt.ToTxAttempt(&attempt)
 	attempts := []TxAttempt{attempt}
-	err := o.preloadTxesAtomic(attempts)
+	err := o.preloadTxesAtomic(ctx, attempts)
 	return &attempts[0], err
 }
 
 // FindTxAttemptsByTxIDs returns a list of attempts by ETH Tx IDs
-func (o *evmTxStore) FindTxAttemptsByTxIDs(ids []int64) ([]TxAttempt, error) {
+func (o *evmTxStore) FindTxAttemptsByTxIDs(ctx context.Context, ids []int64) ([]TxAttempt, error) {
 	sql := `SELECT * FROM evm.tx_attempts WHERE eth_tx_id = ANY($1)`
 	var dbTxAttempts []DbEthTxAttempt
-	if err := o.q.Select(&dbTxAttempts, sql, ids); err != nil {
+	if err := o.q.SelectContext(ctx, &dbTxAttempts, sql, ids); err != nil {
 		return nil, err
 	}
 	return dbEthTxAttemptsToEthTxAttempts(dbTxAttempts), nil
 }
 
-func (o *evmTxStore) FindTxByHash(hash common.Hash) (*Tx, error) {
+func (o *evmTxStore) FindTxByHash(ctx context.Context, hash common.Hash) (*Tx, error) {
 	var dbEtx DbEthTx
-	err := o.q.Transaction(func(tx pg.Queryer) error {
+	err := o.Transaction(ctx, func(orm *evmTxStore) error {
 		sql := `SELECT evm.txes.* FROM evm.txes WHERE id IN (SELECT DISTINCT eth_tx_id FROM evm.tx_attempts WHERE hash = $1)`
-		if err := tx.Get(&dbEtx, sql, hash); err != nil {
+		if err := orm.q.GetContext(ctx, &dbEtx, sql, hash); err != nil {
 			return pkgerrors.Wrapf(err, "failed to find eth_tx with hash %d", hash)
 		}
 		return nil
-	}, pg.OptReadOnlyTx())
```

Comment:
Do we just drop this option?

Reply:
Read only transaction are supported by `sqlutil.Transact`. You shouldn't have to change any behavior.

Reply:
I think I can pass options to restore the read only behaviour